test/autorest/generated/integergroup/models.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package integergroup
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"net/http"
+	"time"
+)
+
+type Error struct {
+	Message *string `json:"message,omitempty"`
+	Status  *int32  `json:"status,omitempty"`
+}
+
+func newError(resp *azcore.Response) error {
+	err := Error{}
+	if err := resp.UnmarshalAsJSON(&err); err != nil {
+		return err
+	}
+	return err
+}
+
+func (e Error) Error() string {
+	msg := ""
+	if e.Message != nil {
+		msg += fmt.Sprintf("Message: %v\n", *e.Message)
+	}
+	if e.Status != nil {
+		msg += fmt.Sprintf("Status: %v\n", *e.Status)
+	}
+	if msg == "" {
+		msg = "missing error info"
+	}
+	return msg
+}
+
+// Int32Response is the response envelope for operations that return a int32 type.
+type Int32Response struct {
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+	Value       *int32
```

Comment:
Note that this codegen is incorrect.  Since the type isn't annotated with `x-nullable` this shouldn't be pointer-to-type and thus fail when receiving a `null` JSON payload.

Reply:
It will require an update to the swagger and test server.

Reply:
Ok will create a PR with those changes in the testserver